File: javascript/two_sum.js

function twoSum(numbs, target) {
    let lengthOfNumbs = numbs.length;
    let map = new Map();
    for (let i = 0; i < lengthOfNumbs; i++) {
        if (map.has(numbs[i])) {
            let numberNeeded = map.get(numbs[i]);
            return [numberNeeded, i];
        } else {
            let targetSubtract = target - numbs[i];
            map.set(targetSubtract, i);
        }
    }
    return [];
}

function test1() {
    const array = [2, 7, 11, 15];
    const target = 9;
    console.log(twoSum(array, target));
}

function test2() {
    const array = [3, 2, 4];
    const target = 6;
    console.log(twoSum(array, target));
}

function test3() {
    const array = [3, 3];
    const target = 6;
    console.log(twoSum(array, target));
}

function display() {
    test1();
    test2();
    test3();
}
display();

File: java/src/TwoSum.java
import java.util.Arrays;
import java.util.HashMap;

public class TwoSum {

    public static int[] twoSum(int[] numbs, int target){
        int lengthOfNumbs = numbs.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < lengthOfNumbs; i++){
            if (map.containsKey(numbs[i])){
                int numberNeeded = map.get(numbs[i]);
                return new int[]{numberNeeded, i};
            } else {
                int targetSubtract = target - numbs[i];
                map.put(targetSubtract, i);
            }
        }
        return new int[2];
    }

    public static void test1(){
        int[] numbs = new int[]{2,7,11,15};
        int target = 9;
        String results = Arrays.toString(twoSum(numbs, target));
        System.out.println(results);
    }

    public static void test2(){
        int[] numbs = new int[]{3,2,4};
        int target = 6;
        String results = Arrays.toString(twoSum(numbs, target));
        System.out.println(results);
    }

    public static void test3(){
        int[] numbs = new int[]{3,3};
        int target = 6;
        String results = Arrays.toString(twoSum(numbs, target));
        System.out.println(results);
    }

    public static void main(String[] args) {
        test1();
        test2();
        test3();
    }
}
